lib/everypolitician/popolo/collection.rb

```diff
@@ -31,8 +31,14 @@ def find_by(attributes = {})
       end
 
       def where(attributes = {})
-        select do |object|
-          attributes.all? { |k, v| object.send(k) == v }
+        index_attributes(attributes)
+        attributes.keys.map { |k| @indexes[k.to_sym][attributes[k]] }.reduce(:&) || []
+      end
+
+      def index_attributes(attributes = {})
```

Comment:
this should probably be in a `private` section, as I doubt it should be part of the public interface.